### PR TITLE
Bring hotkey strings in line with visible menu strings

### LINKED ISSUE #3604: FontForge deletes hotkeys if no font opened; otherwise ignores them
I have the following in `/home/fred/.config/fontforge/hotkeys`:
```XResources
fontforge.CharView.Hotkey.Tool.SpiroG2:1
fontforge.CharView.Hotkey.Tool.SpiroG4:2
fontforge.CharView.Hotkey.Tool.SpiroCorner:3
fontforge.CharView.Hotkey.Tool.SpiroLeft:4
fontforge.CharView.Hotkey.Tool.SpiroRight:5
```

If I run `fontforge -open` and immediately close it, it deletes the file!

Even if I do open a font, these hotkeys don't work. I've tried as well `fontforge.CharView.Menu.Point.Tools.G2` and `CharView.Menu.Point.Tools.G2`, mentioned in `fontforge/share/default-n`, and those don't work either.

### PR DESCRIPTION
**Pull request by skef**
This update brings the hotkey strings in line with the menu strings. Some of the more important entries in the `default` hotkey file already had these values (and therefore didn't work before). Others have been changed. 

Another option is to eliminate the `defaults` edits and just bring the strings in line with it. However, I think it is preferable to be able to "read off" of the needed string from the menu itself. The vast majority of the strings are already like this. 

This is "breaking" in the sense that people might have files with custom hotkeys that have now changed. However, the edited keys are (for the most part) not very likely to have been customized. It also changes the interface in the sense that it enables some hotkeys that didn't work before, mostly in the Point -> Tools menu. 

I'm also disabling saving of the user's hotkey file, because that process is prone to zeroing out the file in some cases, and there doesn't seem to be a good enough reason for it; it just saves what is marked as `isUserDefined` -- a value set when loading entries from that file. (I'm happy to remove that commit on request/explanation.) 

Closes #3604 

### Types of changes
What types of changes does your code introduce? Check all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
Go over all the following points and check all the boxes that apply. 
If you're unsure about any of these, don't hesitate to ask. We're here to help! Various areas of the codebase have been worked on by different people in recent years, so if you are unfamiliar with the general area you're working in, please feel free to chat with people who have experience in that area. See the list [here](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md#people-to-ask).
- [x] My code follows the code style of this project found [here](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md#coding-style).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

**Comment by ctrlcctrlv** [comment]
Fantastic work!&mdash;I **approve** of this pull, and especially invite the change made to ` fontforgeexe/startui.c ` as necessary.

**Comment by ctrlcctrlv** [comment]
@skef I've just noticed something. Is there a reason you've decided to give the Spiro points `Ctrl+1`, `Ctrl+2` etc rather than `1`, `2` etc., other than that's how it was in the broken original?

**Comment by skef** [comment]
I don't believe those are Control-modified -- it's just the bare characters. And no, it's just the consequence of the assigned hotkeys working now.

File: share/default [hunk 1]
```diff
@@ -105,8 +105,8 @@ CharView.Menu.Point.HVCurve:Ctrl+5
 +CharView.Menu.Point.G2 Curve:Ctrl+1
 +CharView.Menu.Point.G4 Curve:Ctrl+2
 +CharView.Menu.Point.Corner:Ctrl+3
-+CharView.Menu.Point.Prev Constraint:Ctrl+4
-+CharView.Menu.Point.Next Constraint:Ctrl+5
```
Comment (ctrlcctrlv) [inline share/default]: It is not the bare numbers @skef&hellip;

Comment (skef) [reply share/default]: Those are different controls directly on the Point menu. They convert selected points between those types. The tools submenu switches between palette tools (a more common operation, so a sensibly simpler key combo).

**Comment by skef** [comment]
(De)activate Spiro was still wrong, so I made it a bare "S". I imagine that might piss off some people, but round-trips through Sprio that don't modify are "safe", right?

**Comment by ctrlcctrlv** [comment]
I use zero, but yes they are safe.

**Comment by skef** [comment]
Put this on hold for now.

**Comment by ctrlcctrlv** [comment]
Why? What's wrong with it?

**Comment by skef** [comment]
Nothing; it's just probably going to be superseded.

**Comment by jtanx** [comment]
Generally looks ok to me. Was it ever possible to edit hotkeys interactively? Otherwise I'm not sure either why there's a need to save the hotkeys file

**Comment by skef** [comment]
This is probably best superseded by #3637, but in case that stalls out let's leave it open for a bit.

**Comment by davelab6** [comment]
> Was it ever possible to edit hotkeys interactively?

I don't think so.

**Comment by skef** [comment]
> This is probably best superseded by #3637, but in case that stalls out let's leave it open for a bit.

Prescience!

**Pull request merged**